### ISSUE TITLE
The blocks can't hold well 

### ISSUE THREAD

**Issue by Hestrong** [Xuzhiqian/Tetris#1]
Hello,I appreciate your repo very much.Recently,we are doing a project about a game on the Diligent BASYS2 board.When we downloaded this repository and modified it to our board,there are some errors happend.The blocks can't hold well and all of them disappeared when they arrive at the down edge of the area.Would you tell me whether this is the bug of the repo or maybe we modified it in a wrong way?And how can we solve this problem in a right way?

**Comment by Xuzhiqian** [comment]
Hi ! Thanks very much!  I never thought this repo would be found by someone else, i'm really excited.
I assume you mean Digilent BASYS2 rather than Diligent BASYS2, right? The game works well on my Digilent NEXYS3 board, and i'm sure they don't disappear when getting to the down edge. Could you please show me the part you have modified? #

**Comment by Xuzhiqian** [comment]
If I may ask, are you Chinese? I guess your name is 何强. 
I think it's better if we can use Chinese, because my English is not so good...

**Comment by Hestrong** [comment]
是的。非常感谢能得到你的回复。我们主要是改动了data.v文件里height的值的大小，把20改为了8，并且后面的与height有关的for循环都进行了改动，因为BASYS2的资源不够，是不是由于与height相关的变量太多，我们未完全修改过来？
`parameter a=20;
reg [5:0] lx;
reg [5:0] ly;

	
always @(posedge clk)
begin
    if (0<lcd_x-40 && lcd_x-40<=a)
        lx<=19;
    else if (a<lcd_x-40 && lcd_x-40<=2*a)
        lx<=18;
    else if (2*a<lcd_x-40 && lcd_x-40<=3*a)
        lx<=17;
    else if (3*a<lcd_x-40 && lcd_x-40<=4*a)
        lx<=16;
    else if (4*a<lcd_x-40 && lcd_x-40<=5*a)
        lx<=15;
    else if (5*a<lcd_x-40 && lcd_x-40<=6*a)
        lx<=14;
    else if (6*a<lcd_x-40 && lcd_x-40<=7*a)
        lx<=13;
    else if (7*a<lcd_x-40 && lcd_x-40<=8*a)
        lx<=12;
    else if (8*a<lcd_x-40 && lcd_x-40<=9*a)
        lx<=11;
    else if (9*a<lcd_x-40 && lcd_x-40<=10*a)
        lx<=10;
    else if (10*a<lcd_x-40 && lcd_x-40<=11*a)
        lx<=9;
    else if (11*a<lcd_x-40 && lcd_x-40<=12*a)
        lx<=8;
    else if (12*a<lcd_x-40 && lcd_x-40<=13*a)
        lx<=7;
    else if (13*a<lcd_x-40 && lcd_x-40<=14*a)
        lx<=6;
    else if (14*a<lcd_x-40 && lcd_x-40<=15*a)
        lx<=5;
    else if (15*a<lcd_x-40 && lcd_x-40<=16*a)
        lx<=4;
    else if (16*a<lcd_x-40 && lcd_x-40<=17*a)
        lx<=3;
    else if (17*a<lcd_x-40 && lcd_x-40<=18*a)
        lx<=2;
    else if (18*a<lcd_x-40 && lcd_x-40<=19*a)
        lx<=1;
    else if (19*a<lcd_x-40 && lcd_x-40<=20*a)
        lx<=0;
    else
        lx<=23;

    if (60<lcd_x && lcd_x<=70)
        show_x<=4;
    else if (70<lcd_x && lcd_x<=80)
        show_x<=3;
    else if (80<lcd_x && lcd_x<=90)
        show_x<=2;
    else if (90<lcd_x && lcd_x<=100)
        show_x<=1;
    else
        show_x<=0;
        
    if (0<lcd_y-200 && lcd_y-200<=a)
        ly<=0;
    else if (a<lcd_y-200 && lcd_y-200<=2*a)
        ly<=1;
    else if (2*a<lcd_y-200 && lcd_y-200<=3*a)
        ly<=2;
    else if (3*a<lcd_y-200 && lcd_y-200<=4*a)
        ly<=3;
    else if (4*a<lcd_y-200 && lcd_y-200<=5*a)
        ly<=4;
    else if (5*a<lcd_y-200 && lcd_y-200<=6*a)
        ly<=5;
    else if (6*a<lcd_y-200 && lcd_y-200<=7*a)
        ly<=6;
    else if (7*a<lcd_y-200 && lcd_y-200<=8*a)
        ly<=7;
    else if (8*a<lcd_y-200 && lcd_y-200<=9*a)
        ly<=8;
    else if (9*a<lcd_y-200 && lcd_y-200<=10*a)
        ly<=9;
    else
        ly<=19;
    
    if (460<lcd_y && lcd_y<=470)
        show_y<=1;
    else if (470<lcd_y && lcd_y<=480)
        show_y<=2;
    else if (480<lcd_y && lcd_y<=490)
        show_y<=3;
    else if (490<lcd_y && lcd_y<=500)
        show_y<=4;  
    else
        show_y<=0;
end`
这一段中的有些数值的设定我们还不是很理解，你觉得会不会是这一部分的原因呢？
另外我的邮箱是heqiang1997@gmail.com,通过邮件可能更方便联系。

**Comment by Xuzhiqian** [comment]
说实话我的编码不是很规范，有很多地方为了方便都用了数值直接代替，所以height的改变可能会导致许多问题。这一段是VGA当前扫描坐标到map坐标的一个映射，比如lx=2,ly=3表示VGA扫描到了map[2][3]这一块，之后可以方便地判断这个位置是否需要被染色（即有方块）。类似地，show_x和show_y代表右上角显示下一个方块的相对坐标。这样的话就可以共享同一个绘图逻辑了。

至于你说的方块落地消失，以前我debug时也遇到过，不过已经被修复了，比如S_fall中的这一条：
`if (x==0) next=S_fix;`
但看起来和height没有关系
或者是因为VGA不能正确绘制已经fix的方块，即不能正确绘制map。
从你发过来的代码来看，我还是不能看出对lx赋值的那一块被如何修改...
我的邮箱是xzq@mail.ustc.edu.cn，
你可以发一份你们综合和编译好的整个项目（包括所有工程文件）到我邮箱里，
我可以在自己板子上测试一下。
